src/applications/facility-locator/hooks/useServiceType.jsx

```diff
@@ -0,0 +1,107 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useCallback, useEffect } from 'react';
+import { connectDrupalStaticDataFileVaHealthServices } from 'platform/site-wide/drupal-static-data/source-files/va-health-services/connect';
+
+const termMatcher = (term, hsdatum, index, includes = false) => {
+  if (includes) {
+    return !!hsdatum[index]?.toLowerCase().includes(term) || false;
+  }
+  const returnMatch = hsdatum[index]?.toLowerCase().search(term);
+  if (returnMatch === undefined) {
+    return -1;
+  }
+  return returnMatch;
+};
+
+const matchHelper = (term, hsdatum) => {
+  const regexTerm = new RegExp(term, 'i');
+  const returnMatcher = {
+    nameMatch: termMatcher(regexTerm, hsdatum, 0),
+    akaMatch: termMatcher(regexTerm, hsdatum, 1),
+    commonCondMatch: hsdatum[2].findIndex(commonCond =>
+      commonCond.toLowerCase().startsWith(term.toLowerCase()),
+    ),
+    apiIdMatch: termMatcher(regexTerm, hsdatum, 3),
+    descriptionMatch: termMatcher(term, hsdatum, 10, true),
+    tricareDescriptionMatch: termMatcher(true, hsdatum, 11, true),
+    hsdatum,
+  };
+
+  if (
+    returnMatcher.nameMatch >= 0 ||
```

Comment:
If `termMatcher` (used on line 19) could return a `true`, `false`, `-1` or `{number}`, does it make sense here to check for `>= 0`? Should we make sure the term matcher always returns a number?

Reply:
Updated, all number returns now (-1 is always does not match/does not include) and >=0 for some match (includes always returns 1 if found)